htsget-search/src/htsget/search.rs

```diff
@@ -57,12 +49,14 @@ pub(crate) async fn find_first<T>(
 /// [SearchAll] represents searching bytes ranges that are applicable to all formats. Specifically,
 /// range for the whole file, and the header.
 ///
+/// [S] is the storage type.
+/// [R] is the inner type used for [Reader].
 /// [ReferenceSequence] is the reference sequence type of the format's index.
 /// [Index] is the format's index type.
 /// [Reader] is the format's reader type.
```

Comment:
Shouldn't this reader be named sth else than "Reader"? I always confused it with R :/

Reply:
Could do something like ReaderType or ReaderInner like with your original code?

Reply:
Yeah, that'd be a bit clearer I reckon. Great job with this PR Marko!!